frontend/packages/vsphere-plugin/src/components/VSphereConnectionModal.tsx

```diff
@@ -0,0 +1,170 @@
+import * as React from 'react';
+import {
+  Modal,
+  ModalVariant,
+  Button,
+  Alert,
+  AlertVariant,
+  AlertActionLink,
+  Stack,
+  StackItem,
+} from '@patternfly/react-core';
+import { useTranslation } from 'react-i18next';
+import { HealthState } from '@console/dynamic-plugin-sdk';
+import { useK8sModel } from '@console/dynamic-plugin-sdk/src/api/core-api';
+import { useConnectionFormContext } from './ConnectionFormContext';
+import { InProgress } from './InProgress';
+import { persist } from './persist';
+import { VSphereConnectionProps } from './types';
+import { VSphereConnectionForm } from './VSphereConnectionForm';
+import { VSphereOperatorStatuses } from './VSphereOperatorStatuses';
+import './VSphereConnectionModal.css';
+
+export const VSphereConnectionModal: React.FC<VSphereConnectionProps> = (params) => {
+  const { t } = useTranslation();
+  const [isModalOpen, setModalOpen] = React.useState(true);
+
+  const [SecretModel] = useK8sModel({ group: 'app', version: 'v1', kind: 'Secret' });
+  const [ConfigMapModel] = useK8sModel({ group: 'app', version: 'v1', kind: 'ConfigMap' });
+
+  const [isSaving, setIsSaving] = React.useState(false);
+  const [error, setError] = React.useState<string>();
+
+  const {
+    vcenter,
+    username,
+    password,
+    datacenter,
+    defaultdatastore,
+    folder,
+    isDirty,
+    setDirty,
+  } = useConnectionFormContext();
+
+  const formId = 'vsphere-connection-modal-form';
+
+  const onClose = () => {
+    setModalOpen(false);
+
+    // hide popup
+    params.hide && params.hide();
+  };
+
+  const onSave = () => {
+    setIsSaving(true);
+    const doItAsync = async () => {
+      setError('');
+
+      const errorMsg = await persist(
+        t,
+        { SecretModel, ConfigMapModel },
+        {
+          vcenter,
+          username,
+          password,
+          datacenter,
+          defaultdatastore,
+          folder,
+        },
+        params.cloudProviderConfig,
+      );
+
+      // Done
+      setIsSaving(false);
+
+      if (errorMsg) {
+        setError(errorMsg);
+        setIsSaving(false);
```

Comment:
this is unnecessary as its already called on line 73
```suggestion
```

Reply:
right